src/Fixer/ConstantNotation/NativeConstantInvocationFixer.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ConstantNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class NativeConstantInvocationFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Add leading `\` before constant invocation of internal constant to speed up resolving. Constant name match is case-sensitive.',
+            [
+                new CodeSample('<?php var_dump(PHP_VERSION, M_PI);'.PHP_EOL),
+                new CodeSample(
+                    '<?php var_dump(PHP_VERSION, M_PI);'.PHP_EOL,
+                    [
+                        'include' => [
+                            'M_PI',
+                        ],
+                    ]
+                ),
+                new CodeSample(
+                    '<?php var_dump(PHP_VERSION, M_PI);'.PHP_EOL,
+                    [
+                        'exclude' => [
+                            'M_PI',
+                        ],
+                    ]
+                ),
+            ],
+            null,
+            'Risky when any of the constants are namespaced or overridden.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_STRING);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $constantNames = $this->getConstantNames();
```

Comment:
let us override `configure` and store `constantNames` in property. that way, it will be calculated once instead of per file

Reply:
actually, i would convert it even to dict `['M_PI' => true, ...]`, so lookup if given const is supposed to be slashed would be `O(1)` (`isset($constantNames[$foo])`) instead of `O(n)` (`in_array($foo, $constantNames, true)`)


https://3v4l.org/qjKCq

WDYT?

Reply:
I know the isset/in_array thing, I started it this way because of casing.

So let me ask you one thing, linked to NULL/TRUE/FALSE comment: how should I handle casing?
Because `M_PI !== m_pi`: https://3v4l.org/cmQCG

Reply:
both `isset` and `in_array` are case sensitive.
I don't see that you do casing manipulation.

what is the issue ?